packages/frontend/src/components/composer/Composer.tsx

```diff
@@ -134,6 +134,45 @@ const Composer = forwardRef<
     [chatId, updateDraftText]
   )
 
+  const onArrowUpWhenEmpty = useCallback(async () => {
+    if (messageEditing.isEditingModeActive) {
+      return
+    }
+    try {
+      // Get message list items for the chat
+      const messageListItems = await BackendRemote.rpc.getMessageListItems(
```

Comment:
This call and the next BackendRemote.rpc.getMessages call can be avoided by passing the messageCache from MessageListAndComposer to Composer as a prop.

```
messageCache={messageListState.messageCache}
```
and then use the messageCache similar to Line 451 in [useDraft](https://github.com/deltachat/deltachat-desktop/blob/main/packages/frontend/src/hooks/chat/useDraft.ts#L451)

```suggestion
     // Dummy code!
      const messageIds = Object.keys(messageListState.messageCache)
      .map(Number)
      
    const message = messageListState.messageCache[messageIds.length - 1]
```